org.eclipse.jdt.core/model/org/eclipse/jdt/core/compiler/CompilationParticipant.java

```diff
@@ -143,6 +143,22 @@ public void processAnnotations(BuildContext[] files) {
 	// do nothing by default
 }
 
+/**
+ * Notifies this participant about a finished class compilation. This
+ * notification occurs right before the compiled class is written out to disk.
+ * This allows to perform arbitrary byte code manipulation on the generated
+ * class.
+ *
+ * @param file BuildContext of the generated class
+ * @param bytes byte representation of the generated class
+ * @return byte array with length > 0 representing the modified class or byte[0]
+ *         if no modification was applied.
+ * @since 3.34
+ */
+public byte[] postProcess(BuildContext file, byte[] bytes) {
+	return new byte[0];
```

Comment:
do we ususally have contstants for such magic objects?

Reply:
One can create one in this class, something like `NO_CHANGE`.

Reply:
I'm thinking about `Optional<byte[]>` which would communicate the intend arguably better.